src/coreclr/jit/importer.cpp

```diff
@@ -4238,13 +4238,21 @@ GenTree* Compiler::impImportStaticFieldAccess(CORINFO_RESOLVED_TOKEN* pResolvedT
         case CORINFO_FIELD_STATIC_RELOCATABLE:
         {
 #ifdef FEATURE_READYTORUN
-            assert(pFieldInfo->fieldLookup.accessType == InfoAccessType::IAT_VALUE);
             assert(fieldKind == FieldSeq::FieldKind::SimpleStatic);
             assert(innerFldSeq != nullptr);
 
-            GenTree* baseAddr = gtNewIconHandleNode((size_t)pFieldInfo->fieldLookup.addr, GTF_ICON_STATIC_HDL);
-            GenTree* offset   = gtNewIconNode(pFieldInfo->offset, innerFldSeq);
-            op1               = gtNewOperNode(GT_ADD, TYP_I_IMPL, baseAddr, offset);
+            size_t fldAddr = (size_t)pFieldInfo->fieldLookup.addr;
+            if (pFieldInfo->fieldLookup.accessType == IAT_VALUE)
+            {
+                op1 = gtNewIconHandleNode(fldAddr, GTF_ICON_STATIC_HDL);
+            }
+            else
+            {
+                assert(pFieldInfo->fieldLookup.accessType == IAT_PVALUE);
+                op1 = gtNewIndOfIconHandleNode(TYP_BYREF, fldAddr, GTF_ICON_STATIC_ADDR_PTR, true);
```

Comment:
Just to double check - typing this a `TYP_BYREF` and retyping it as `TYP_I_IMPL` when creating the `GT_ADD` below won't cause any issues?

Could we just type it as TYP_I_IMPL here? If I'm reading it right we're assuming it's pinned anyway.

Reply:
Ah, right, `TYP_I_IMPL` works here to for the static base address (and as we discussed inhttps://github.com/dotnet/runtime/pull/79709 it doesn't lead to constant folding) and won't generate unnecessary gc info. The root `GT_IND` node has `TYP_REF` for gc refs so everything should be fine.